src/v/kafka/server/handlers/topics/types.h

```diff
@@ -59,7 +59,15 @@ static constexpr std::string_view topic_property_data_policy_script_name
   = "redpanda.datapolicy.script.name";
 
 // Kafka topic properties that is not relevant for Redpanda
-static constexpr std::string_view allowlist_topic_noop_confs[20] = {
+// Or cannot be altered with kafka alter handler
+static constexpr std::string_view allowlist_topic_noop_confs[23] = {
+  // Cannot be altered in handle
+  "partition_count",
+  "replication_factor",
+  // Invalid name from describe
+  "redpanda.datapolicy",
+
```

Comment:
Note: I think it should be fine for now but having a way to annotate the scope of each property would be nice to have just to make this better in the future.

Reply:
we can also make this a `ctre` ... something like this https://godbolt.org/z/5PcbEWz8s